### ISSUE TITLE
[FEA] enhancement: not allow getData from ColumnVector of nested type

### ISSUE THREAD

**Issue by wjxiz1992** [rapidsai/cudf#9497]
**Is your feature request related to a problem? Please describe.**
When getData from a ColumnVector of nested type such as LIST, the address will be null, and that will trigger a NPE.

**Describe the solution you'd like**
When calling getData API from a ColumnVector, check the data type, if it's nested type, throw exception to inform that no data could be get from a nested type ColumnVector.

**Additional context**

```scala
// assume this columnVector is of nested type : LIST
val data = columnVector.getData  // null
val addr = childData.getAddress
```

**Comment by GregoryKimball** [comment]
@ttnghia is this still needed?

**Comment by ttnghia** [comment]
Yes this is still needed. Not a bug but having this can improve stability/reduce hidden bugs. Working on this should be very simple, just adding 1 LOC. I will take care of this.

**Comment by jlowe** [comment]
Agree this is pretty minor.  The Javadoc clearly states that it returns null if there's no buffer.  We could change the behavior to throw on no buffer instead of returning null.  I don't know how many other places are already coded against the existing functionality, handling the null themselves, and would break with a thrown exception.  Not sure this change is worth the risk of breakage.
